### PR TITLE
Test multiple set sizes in testLargeIn

### PR DESCRIPTION
**Pull request by losipiuk**
It is possible that connector has mechanisms which ensure that very
large value sets (>5000) work fine, yet it would still fail for smaller,
yet not tiny sets. Adding coverage for that.

### CODE REVIEW

**Review by sopel39 — APPROVED** [review]
Does it increase test runtime?

**Comment by losipiuk** [comment]
> Does it increase test runtime?

Surely :)

Lets see on CI how much. Locally I tested it for some arbitrary connector and it was not substantial (~1min)

File: presto-memsql/src/test/java/io/prestosql/plugin/memsql/TestMemSqlDistributedQueries.java [hunk 1]
```diff
@@ -130,7 +130,7 @@ protected boolean isColumnNameRejected(Exception exception, String columnName, b
     }
 
     @Override
-    public void testLargeIn()
+    public void testLargeIn(int valuesCount)
```
Comment (findepi) [inline presto-memsql/src/test/java/io/prestosql/plugin/memsql/TestMemSqlDistributedQueries.java:133]: We should be able to enable the test now for memsql.

(make a PR from inside the repo to have memsql tests run)

File: presto-phoenix/src/test/java/io/prestosql/plugin/phoenix/TestPhoenixDistributedQueries.java [hunk 2]
```diff
@@ -75,7 +75,7 @@ protected TestTable createTableWithDefaultColumns()
     }
 
     @Override
-    public void testLargeIn()
+    public void testLargeIn(int valuesCount)
```
Comment (findepi) [inline presto-phoenix/src/test/java/io/prestosql/plugin/phoenix/TestPhoenixDistributedQueries.java:78]: We should be able to enable the test now for phoenix, provided that @sopel39 's capping works for phoenix too.

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestQueries.java [hunk 3]
```diff
@@ -652,10 +653,10 @@ public void testIn()
...
-    @Test
-    public void testLargeIn()
+    @Test(dataProvider = "largeInValuesCount")
+    public void testLargeIn(int valuesCount)
     {
-        String longValues = range(0, 5000)
+        String longValues = range(0, valuesCount)
                 .mapToObj(Integer::toString)
                 .collect(joining(", "));
         assertQuery("SELECT orderkey FROM orders WHERE orderkey IN (" + longValues + ")");
```
Comment (findepi) [inline presto-testing/src/main/java/io/prestosql/testing/AbstractTestQueries.java:662]: would it be possible to calc expected values without h2? would that improve test runtime?

like:
```
assertThat(query("....."))
  .matches( materialized result )
```



(treat this optional)

**Review by findepi — APPROVED** [review]

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestQueries.java (hunk 3 above)
Comment (losipiuk) [reply presto-testing/src/main/java/io/prestosql/testing/AbstractTestQueries.java:662]: I will skip this one. Materialized rows for "not in" are kinda painful.

File: presto-memsql/src/test/java/io/prestosql/plugin/memsql/TestMemSqlDistributedQueries.java (hunk 1 above)
Comment (losipiuk) [reply presto-memsql/src/test/java/io/prestosql/plugin/memsql/TestMemSqlDistributedQueries.java:133]: https://github.com/prestosql/presto/pull/6312

**Comment by losipiuk** [comment]
> LGTM'd but please check the runtime times on CI before merging.

Comparing to master there seem to be a cost of couple minutes for relevant jobs (3-15 minutes extra). Sometimes this PR is also faster than master but I guess it comes from huge variance. Still none of the test jobs for which this change matter became the longest one in the workflow.

I think this is ok to merge.  I can also make list of tested values slightly shorter. 
Say 100, 500, 2000, 5000. WDYT @findepi

**Comment by sopel39** [comment]
> I think this is ok to merge. I can also make list of tested values slightly shorter.
Say 100, 500, 2000, 5000. WDYT @findepi

What was the case where you observed a failure on smaller list? What was the reason?

**Comment by sopel39** [comment]
> Comparing to master there seem to be a cost of couple minutes for relevant jobs (3-15 minutes extra)

This seems like a lot for a single test.

Would 100, 1000, 5000 be sufficient?

**Comment by losipiuk** [comment]
> > Comparing to master there seem to be a cost of couple minutes for relevant jobs (3-15 minutes extra)
> 
> This seems like a lot for a single test.
> 
> Would 100, 1000, 5000 be sufficient? 

I left 100, 200, 1000, 5000 as I would like to have lower end covered more dense (also the lower values are cheaper, so it will not add a lot to current 5k).

> Also does it mean that some connectors still have compaction limit too high?

I think the cost is within parser, analyzer and planner. But I would need to profile to verify.

**Comment by sopel39** [comment]
@losipiuk this is most likely related too https://github.com/prestosql/presto/issues/5950

**Pull request merged**